protocols/cifs/src/main/java/org/opennms/netmgt/poller/monitors/JCifsMonitor.java

```diff
@@ -216,6 +223,32 @@ public boolean accept(SmbFile smbFile, String s) throws SmbException {
         return serviceStatus;
     }
 
+    private void setFinalStatic(final Field finalStaticField, final Object value) throws NoSuchFieldException, IllegalAccessException {
+        finalStaticField.setAccessible(true);
+
+        final Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(finalStaticField, finalStaticField.getModifiers() & ~Modifier.FINAL);
+
+        finalStaticField.set(null, value);
+    }
+
+    /**
+     * Due to limitations of the JCifs library we must use this hack to modify the timeouts. Since the timeouts are set
+     * in a static fields only global timeouts will work reliably.
+     *
+     * @param t the timeout to be set
+     * @throws NoSuchFieldException
+     * @throws ClassNotFoundException
+     * @throws IllegalAccessException
+     */
+    private void setJCifsTimeouts(int t) throws NoSuchFieldException, ClassNotFoundException, IllegalAccessException {
+        final Class<?> clazz = Class.forName("jcifs.smb.SmbConstants");
+        setFinalStatic(clazz.getField("SO_TIMEOUT"), t);
+        setFinalStatic(clazz.getField("CONN_TIMEOUT"), t);
+        setFinalStatic(clazz.getField("DEFAULT_RESPONSE_TIMEOUT"), t + 5000);
```

Comment:
Can we add a comment, why we add `5000` to it? Also `t` should be renamed to `timeoutInSeconds` or `timeoutInMilliseconds`, depending on what unit it is representing.

Reply:
done